src/metabase/api/database.clj

```diff
@@ -304,8 +304,8 @@
               add-expanded-schedules
               (get-database-hydrate-include include))
 
-    api/*is-superuser?* ; for admins, expand inferred secret values in db-details
-    secret/admin-expand-db-details-inferred-secret-values))
+    mi/can-write?
```

Comment:
should this be `mi/can-read?

Reply:
also, we need to update the docs about renaming the `admin-expand-db-details-inferred-secret-values` func

Reply:
No, this should only be done for users who have write permissions (i.e. can edit DB details), not for any users who can read the DB details